scripts/cc-cli/src/utils/account.ts

```diff
@@ -14,69 +15,83 @@ export function initECDSAKeyringPairFromPK(pk: string) {
+  options: OptionValues,
+): Promise<KeyringPair> {
   try {
-    return await getSeedFromEnvOrPrompt("CC_STASH_SEED", "stash", interactive);
+    return await initKeyringFromEnvOrPrompt("CC_STASH_SEED", "stash", options);
   } catch (e) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
-export async function getControllerSeedFromEnvOrPrompt(interactive: boolean) {
+
+export async function initControllerKeyring(
+  options: OptionValues,
+): Promise<KeyringPair> {
   try {
-    return await getSeedFromEnvOrPrompt(
+    return await initKeyringFromEnvOrPrompt(
       "CC_CONTROLLER_SEED",
       "controller",
-      interactive,
+      options,
     );
   } catch (e) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
-export async function getCallerSeedFromEnvOrPrompt(interactive: boolean) {
+
+export async function initCallerKeyring(
+  options: OptionValues,
+): Promise<KeyringPair> {
   try {
-    return await getSeedFromEnvOrPrompt("CC_SEED", "caller", interactive);
+    return await initKeyringFromEnvOrPrompt("CC_SEED", "caller", options);
   } catch (e) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
 
-async function getSeedFromEnvOrPrompt(
-  envVar = "CC_SEED",
-  accountRole = "caller",
-  interactive = true,
-) {
+export async function initKeyringFromEnvOrPrompt(
+  envVar: string,
+  accountRole: string,
+  options: OptionValues,
+): Promise<KeyringPair> {
+  const interactive = options.input;
+  const ecdsa = options.ecdsa;
+  envVar = ecdsa ? "CC_PK" : envVar;
+  const inputName = ecdsa ? "private key" : "seed phrase";
+  const validateInput = ecdsa ? () => true : mnemonicValidate;
+  const generateKeyring = ecdsa ? initECDSAKeyringPairFromPK : initKeyringPair;
+
   if (!interactive && !process.env[envVar]) {
     throw new Error(
-      `Error: Must specify a seed phrase for the ${accountRole} account in the environment variable ${envVar} or use an interactive shell.`,
+      `Error: Must specify a ${inputName} for the ${accountRole} account in the environment variable ${envVar} or use an interactive shell.`,
     );
   }
 
   if (typeof process.env[envVar] === "string") {
-    const seedFromEnv = process.env[envVar];
-    if (mnemonicValidate(seedFromEnv!)) {
-      return seedFromEnv;
+    if (validateInput(process.env[envVar]!)) {
+      return generateKeyring(process.env[envVar]!);
     } else {
       throw new Error(
         `Error: Seed phrase provided in environment variable ${envVar} is invalid.`,
       );
     }
   } else if (interactive) {
-    const seedPromptResult = await prompts([
+    const promptResult = await prompts([
       {
         type: "password",
         name: "seed",
-        message: `Specify a seed phrase for the ${accountRole} account`,
-        validate: (seed) => mnemonicValidate(seed),
+        message: `Specify a ${inputName} for the ${accountRole} account`,
+        validate: (input) => validateInput(input),
```

Comment:
During interactive mode we can still use 2 different Stash & Controller accounts.